tests/pq/test_pipeline.py

```diff
@@ -0,0 +1,87 @@
+import pytest
+
+import psycopg
+from psycopg import pq
+
+pytestmark = pytest.mark.libpq(">= 14")
+
+
+def test_work_in_progress(pgconn):
+    assert not pgconn.nonblocking
+    pgconn.enter_pipeline_mode()
+    pgconn.send_query_params(b"select $1", [b"1"])
+    with pytest.raises(
+        psycopg.OperationalError, match="cannot exit pipeline mode"
+    ):
+        pgconn.exit_pipeline_mode()
+
+
+def test_multi_pipelines(pgconn):
+    pgconn.enter_pipeline_mode()
+    pgconn.send_query_params(b"select $1", [b"1"])
+    pgconn.pipeline_sync()
+    pgconn.send_query_params(b"select $1", [b"2"])
+    pgconn.pipeline_sync()
+    result1 = pgconn.get_result()
+    assert result1 is not None
+    assert result1.status == pq.ExecStatus.TUPLES_OK
+    assert pgconn.get_result() is None
+    result2 = pgconn.get_result()
+    assert result2 is not None
+    assert result2.status == pq.ExecStatus.PIPELINE_SYNC
+    result3 = pgconn.get_result()
+    assert result3 is not None
+    assert result3.status == pq.ExecStatus.TUPLES_OK
+    assert pgconn.get_result() is None
+    result4 = pgconn.get_result()
+    assert result4 is not None
+    assert result4.status == pq.ExecStatus.PIPELINE_SYNC
+    assert pgconn.pipeline_status() == pq.PipelineStatus.ON
+    pgconn.exit_pipeline_mode()
+    assert pgconn.pipeline_status() == pq.PipelineStatus.OFF
+    assert result1.get_value(0, 0) == b"1"
+    assert result3.get_value(0, 0) == b"2"
+
+
+@pytest.fixture
+def table(pgconn):
+    tablename = "pipeline"
+    pgconn.exec_(f"create table {tablename} (s text)".encode("ascii"))
+    yield tablename
+    pgconn.exec_(f"drop table if exists {tablename}".encode("ascii"))
+
+
+def test_pipeline_abort(pgconn, table):
+    pgconn.enter_pipeline_mode()
+    pgconn.send_query_params(b"insert into pipeline values ($1)", [b"1"])
+    pgconn.send_query_params(b"select no_such_function($1)", [b"1"])
+    pgconn.send_query_params(b"insert into pipeline values ($1)", [b"2"])
+    pgconn.pipeline_sync()
+    pgconn.send_query_params(b"insert into pipeline values ($1)", [b"3"])
+    pgconn.pipeline_sync()
+    r = pgconn.get_result()
```

Comment:
Could you please make this test a bit more readable by dividing the following lines in stanzas and comment to which one of the statements above they match? Thank you :)

Reply:
Done.